src/Middleware/MiddlewareInterface.php

```diff
@@ -38,9 +38,19 @@
 use Google\ApiCore\ServerStream;
 use Google\ApiCore\BidiStream;
 
+/**
+ * Middlewares should take a MiddlewareInterface as a constructor argument
+ * {@see Google\ApiCore\Middleware\ResponseMetadataMiddleware} which represents
+ * the next middleware in the chain. This will be invoked as part of this
+ * middleware's invoke method, and the result returned.
+ */
 interface MiddlewareInterface
 {
     /**
+     * Modify or observe the request and response parameters.
+     * The returned value should be the result of the next middleware in the
```

Comment:
```suggestion
     * The returned value must include the result of the next MiddlewareInterface invocation in the
```

Stricter language with "must", and slightly looser language with "include the result", in case **this** MiddlewareInterface wants to modify it in some way.